allensdk/brain_observatory/behavior/behavior_ophys_experiment.py

```diff
@@ -1,33 +1,241 @@
+                 eye_tracking_table: Optional[EyeTrackingTable],
+                 eye_tracking_rig_geometry: Optional[EyeTrackingRigGeometry],
+                 date_of_acquisition: DateOfAcquisition):
+        super().__init__(
+            behavior_session_id=behavior_session._behavior_session_id,
+            licks=behavior_session._licks,
+            metadata=behavior_session._metadata,
+            raw_running_speed=behavior_session._raw_running_speed,
+            rewards=behavior_session._rewards,
+            running_speed=behavior_session._running_speed,
+            running_acquisition=behavior_session._running_acquisition,
+            stimuli=behavior_session._stimuli,
+            stimulus_timestamps=behavior_session._stimulus_timestamps,
+            task_parameters=behavior_session._task_parameters,
+            trials=behavior_session._trials,
+            date_of_acquisition=date_of_acquisition
+        )
+
+        self._metadata = metadata
+        self._projections = projections
+        self._ophys_timestamps = ophys_timestamps
+        self._cell_specimens = cell_specimens
+        self._motion_correction = motion_correction
+        self._eye_tracking = eye_tracking_table
+        self._eye_tracking_rig_geometry = eye_tracking_rig_geometry
+
+    def to_nwb(self) -> NWBFile:
+        nwbfile = super().to_nwb(add_metadata=False)
+
+        self._metadata.to_nwb(nwbfile=nwbfile)
+        self._projections.to_nwb(nwbfile=nwbfile)
+        self._cell_specimens.to_nwb(nwbfile=nwbfile,
+                                    ophys_timestamps=self._ophys_timestamps)
+        self._motion_correction.to_nwb(nwbfile=nwbfile)
+        self._eye_tracking.to_nwb(nwbfile=nwbfile)
+        self._eye_tracking_rig_geometry.to_nwb(nwbfile=nwbfile)
+
+        return nwbfile
+    # ==================== class and utility methods ======================
+
+    @classmethod
+    def from_lims(cls,
+                  ophys_experiment_id: int,
+                  eye_tracking_z_threshold: float = 3.0,
+                  eye_tracking_dilation_frames: int = 2,
+                  events_filter_scale: float = 2.0,
+                  events_filter_n_time_steps: int = 20,
+                  exclude_invalid_rois=True,
+                  skip_eye_tracking=False) -> \
+            "BehaviorOphysExperiment":
+        """
+        Parameters
+        ----------
+        ophys_experiment_id
+        eye_tracking_z_threshold
+            See `BehaviorOphysExperiment.from_nwb`
+        eye_tracking_dilation_frames
+            See `BehaviorOphysExperiment.from_nwb`
+        events_filter_scale
+            See `BehaviorOphysExperiment.from_nwb`
+        events_filter_n_time_steps
+            See `BehaviorOphysExperiment.from_nwb`
+        exclude_invalid_rois
+            Whether to exclude invalid rois
+        skip_eye_tracking
+            Used to skip returning eye tracking data
+        """
+        def _is_multi_plane_session():
```

Comment:
This is personal style. Since `_is_multi_plane_session` is in algorithm in itself, and since it is local to `from_lims`, I placed it as a function within this method. I think it is clearer than just writing the algorithm for determining `is_multi_plane_session_ as lines of code within `from_lims` and writing a comment or something above it saying what those lines do.